### PR TITLE
Upgrade to Panda v7 - support key rotation

### PR DESCRIPTION
**Pull request by rtyley**
This upgrades Panda from v5 to v7, allowing us to use [key rotation](https://github.com/guardian/pan-domain-authentication/pull/150).

As Atom Workshop is pretty standard user of Panda, the upgrade is pretty simple:

* Panda v6 requirements:
  * https://github.com/guardian/pan-domain-authentication/pull/155 requires `panDomainSettings` is a `val`, not a `def`
  * https://github.com/guardian/pan-domain-authentication/pull/151 introduced the new `S3BucketLoader` abstraction, which simplifies constructing a `PanDomainAuthSettingsRefresher` and means that Panda is no longer _tied_ to AWS SDK v1 - an alternative AWS SDK v2 implementation of `S3BucketLoader` could be introduced.

This has been [successfully deployed](https://riffraff.gutools.co.uk/deployment/view/86b7ccf9-7117-439c-9c8c-0edbcfb351b0) to https://atomworkshop.code.dev-gutools.co.uk/ at 55de0610d0fe034f7a7462d31bf991dbb6318e00, which is using https://github.com/guardian/pan-domain-authentication/pull/151/commits/4c87946a113b73f23e9e9b683082bdaeb49a5e2d.

## Testing

Using https://atomworkshop.code.dev-gutools.co.uk/ (running Panda with the version from https://github.com/guardian/pan-domain-authentication/pull/151) in the same browser as https://composer.code.dev-gutools.co.uk/ (running standard Panda v4), I've verified that they both accept the same cookie, and don't need to reauth over the top of each-others cookie:

https://github.com/user-attachments/assets/5e2db226-2ab2-4d00-af1c-d601e98c81a2

### CODE REVIEW

**Comment by rtyley** [comment]
Trying out https://github.com/guardian/pan-domain-authentication/pull/150 at https://github.com/guardian/pan-domain-authentication/commit/a6ae5e80a8b0d67c11aa9be70b0f73a97eb1d8c4 on CODE, everything looks good, except that we seem to be getting way more "Panda settings changed" messages in [the logs](https://logs.gutools.co.uk/goto/120023c0-710c-11ef-b481-afe258002fe1) than we were expecting - there should be just 1 per EC2 instance, but we actually see, for just 1 instance, 56, then 64, then 80, for the 3 successive phases of rotation.

**Comment by rtyley** [comment]
Looks like https://github.com/guardian/pan-domain-authentication/commit/6e11a68df89c24c3cc3df4d3b5135c57d067ea45 has fixed the problem, looking at [the logs](https://logs.gutools.co.uk/goto/3351b330-711c-11ef-b481-afe258002fe1):

image

File: app/AppComponents.scala [hunk 1]
```diff
@@ -30,7 +30,7 @@ class AppComponents(context: Context, identity: AppIdentity)
 
     override def controllerComponents: ControllerComponents = AppComponents.this.controllerComponents
 
-    override def panDomainSettings: PanDomainAuthSettingsRefresher = new PanDomainAuthSettingsRefresher(
+    override val panDomainSettings: PanDomainAuthSettingsRefresher = new PanDomainAuthSettingsRefresher(
```
Comment (rtyley) [inline app/AppComponents.scala:33]: Note that this is necessary due to:

* https://github.com/guardian/pan-domain-authentication/pull/155

File: app/AppComponents.scala [hunk 2]
```diff
@@ -30,12 +30,10 @@ class AppComponents(context: Context, identity: AppIdentity)
 
     override def controllerComponents: ControllerComponents = AppComponents.this.controllerComponents
 
-    override def panDomainSettings: PanDomainAuthSettingsRefresher = new PanDomainAuthSettingsRefresher(
+    override val panDomainSettings: PanDomainAuthSettingsRefresher = PanDomainAuthSettingsRefresher(
```
Comment (rtyley) [inline app/AppComponents.scala:33]: The `def` → `val` is necessary due to:

* https://github.com/guardian/pan-domain-authentication/pull/155

File: app/AppComponents.scala [hunk 3]
```diff
@@ -30,12 +30,10 @@ class AppComponents(context: Context, identity: AppIdentity)
...
       domain = config.pandaDomain,
       system = config.pandaSystem,
-      bucketName = "pan-domain-auth-settings",
-      settingsFileKey = s"${config.pandaDomain}.settings",
-      s3Client = AWS.S3Client,
+      S3BucketLoader.forAwsSdkV1(AWS.S3Client, "pan-domain-auth-settings")
```
Comment (rtyley) [inline app/AppComponents.scala:36]: This PR:

* https://github.com/guardian/pan-domain-authentication/pull/151

...introduced the new `S3BucketLoader` abstraction, which simplifies constructing a `PanDomainAuthSettingsRefresher` and means that Panda is no longer _tied_ to AWS SDK v1 - an alternative AWS SDK v2 implementation of `S3BucketLoader` could be introduced.

**Review by Fweddi — APPROVED** [review]

**Pull request merged**